### DISCUSSION TITLE
How to build a separate js file out side of main app bundle?

### DISCUSSION THREAD

**Discussion by guoyunhe** [vercel/next.js#15969] [Help] [answered]
I am using Firebase Messaging which requires service workers as a separate JS file https://github.com/firebase/quickstart-js/blob/master/messaging/firebase-messaging-sw.js

In a next js project , how could I create a script that is not in the main bundle? (still need webpack to bundle required packages)

**Comment by jamesmosier** [comment, marked as answer]
[next-offline](https://github.com/hanford/next-offline) might be a good resource to look towards for doing something similar.. It will require adding some custom webpack configurations to your `next.config.js`

next-offline augments the webpack entires [as seen here](https://github.com/hanford/next-offline/blob/3ea1e13c1ff35ff554f60087fc37e998bfe959d9/packages/next-offline/index.js#L85-L105)
